src/lib/dialog/dialog.ts

```diff
@@ -122,6 +124,20 @@ export class MdDialog {
 
     return state;
   }
+
+  /**
+   * Applies default options to the dialog config.
+   * @param {MdDialogConfig} config Config to be modified.
+   * @returns The new configuration object.
```

Comment:
Are the JSDoc types necessary here? I think the TypeScript types should be enough.